CppCoreGuidelines.md

```diff
@@ -9810,19 +9809,24 @@ Using `unique_ptr` in this way both documents and enforces the function call's r
 * (Simple) Warn if a function takes a `Unique_pointer<T>` parameter by lvalue reference and does not either assign to it or call `reset()` on it on at least one code path. Suggest taking a `T*` or `T&` instead.
 * (Simple) ((Foundation)) Warn if a function takes a `Unique_pointer<T>` parameter by reference to `const`. Suggest taking a `const T*` or `const T&` instead.
 
-### <a name="Rr-sharedptrparam-owner"></a>R.34: Take a `shared_ptr<widget>` parameter to express that a function is part owner
+### <a name="Rr-sharedptrparam-owner"></a>R.34: Take a `shared_ptr<widget>` parameter to express shared ownership
 
 ##### Reason
 
 This makes the function's ownership sharing explicit.
 
 ##### Example, good
 
-    void share(shared_ptr<widget>);            // share -- "will" retain refcount
-
-    void may_share(const shared_ptr<widget>&); // "might" retain refcount
-
-    void reseat(shared_ptr<widget>&);          // "might" reseat ptr
```

Comment:
maybe it's just me but share/may_share/reseat offer a nice contrast. 

also, since pass by-value already paid the cost of incrementing refcount, a good example that makes use of this distinction would be moving out of its parameter, contrasted with may_share which delays the cost of incrementing refcount until the optional code path inside the function where it must do so.

Reply:
Should the examples be shared again?  The current examples are non-member functions which is unusual for shared_ptr.  I thought a constructor would be more typical.

Using std::move is a good point.  I added it to the example.